containers/runtime/php7/run_php7_worker.sh

```diff
@@ -13,6 +13,25 @@
 # limitations under the License.
 
 set -ex
+
+while getopts s: flag; do
+  case "${flag}" in
+  s) input_server_port=${OPTARG} ;;
+  esac
+done
+
+echo "Server port: $input_server_port"
+
 source /usr/local/rvm/scripts/rvm
 export GEM_HOME=/src/workspace/saved/bundle/
-ruby src/ruby/qps/proxy-worker.rb
+
+if [ -z "$input_server_port" ]; then
+  echo "Server port is not set, starting the worker without server port provided"
+  timeout --kill-after=$KILL_AFTER $POD_TIMEOUT ruby src/ruby/qps/proxy-worker.rb \
```

Comment:
I would make sure to put "${KILL_AFTER}" everywhere and so on.

You can run "shellcheck" on your shell script to see what it finds (there is an extension for that in vscode too, not sure about other editors).